maven-release-manager/src/main/java/org/apache/maven/shared/release/DefaultReleaseManager.java

```diff
@@ -304,21 +305,26 @@ private void perform( ReleasePerformRequest performRequest, ReleaseResult result
             ReleaseUtils.buildReleaseDescriptor( performRequest.getReleaseDescriptorBuilder() )
             .getActivateProfiles();
 
-        ReleaseDescriptor releaseDescriptor =
-            loadReleaseDescriptor( performRequest.getReleaseDescriptorBuilder(),
-                                   performRequest.getReleaseManagerListener() );
+        ReleaseDescriptorBuilder builder =
+            loadReleaseDescriptorBuilder( performRequest.getReleaseDescriptorBuilder(),
+                                          performRequest.getReleaseManagerListener() );
 
         if ( specificProfiles != null && !specificProfiles.isEmpty() )
         {
+            List<String> allProfiles = new ArrayList<>();
```

Comment:
I guess this part can result is a different set of activated profiles, but it is not covered by a test. Currently, the build is green and we've seen issues with profiles before, so we really need a higher code coverage here.

Reply:
once again ... this PR fix my previous one, where a Unit Test WAS ADDED

https://github.com/apache/maven-release/pull/50/files#diff-c4d606c66477b937977eccf603f2f04a7b87f464b7439bcb8e4576307792ff2cR709

and this PR changes a little bit this test

Reply:
But if the current build is green, then there is no need to merge this.

Reply:
the current build is green because of the  `new ArrayList( ... )` in the unit test :
```secondBuilder.setActivateProfiles( new ArrayList( Arrays.asList("aProfile", "bProfile") ) );```

Remove it and the build will become red. And IT WAS a MISTAKE to add it, it is false comparing to what happens when using the plugin

This PR fix the code so that this `new ArrayList( ... )` is not needed anymore, and the unit test is representative of the plugin usage

Reply:
Some backgroud is in https://issues.apache.org/jira/browse/MRELEASE-1042 also ...

Reply:
Yes I remember the discussion. @rfscholte you wanted by design that the state of descriptor is immutable ([link](https://github.com/apache/maven-release/pull/56#discussion_r427064656)).
The test case use a new arraylist that is mutable but this is not at runtime. I know because we was forced to fork this repo and make our version of release plugin with a issue (1 year ago!).

Reply:
Here we inject an immutable list - so the test case is in fact wrong.

https://github.com/apache/maven-release/blob/master/maven-release-manager/src/main/java/org/apache/maven/shared/release/config/ReleaseUtils.java#L130-L134

Reply:
@slachiewicz Yes, EXACTLY !!! The test case added in my previous PR is WRONG, and HERE, in THIS PR, I want to fix it ...
Please please, tell me what to do to get this merged and released, my team just NEED this ...

Reply:
@rfscholte please could you finalize this issue?